src/lib.rs

```diff
@@ -11,6 +11,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#![cfg_attr(test, deny(warnings))]
```

Comment:
I don't think CI is guaranteed to run the same test cases with the same configuration in same environment as contributors'. It's normal that local tests succeed but CI fails. What is actually needed is clear fail message and logs.